Kernel/matrixElements.m

```diff
@@ -848,22 +928,30 @@
 	scalarMass=particleMass[[3]];
 
 (*Vector propagators*)
-	vectorPropS=Table[1/(s-i),{i,vectorMass}]//ListToMat;
+	vectorPropS=Table[Prop[s,i],{i,vectorMass}]//ListToMat;
 
 (*Scalar propagators*)
-	scalarPropS=Table[1/(s-i),{i,scalarMass}]//ListToMat;
+	scalarPropS=Table[Prop[s,i],{i,scalarMass}]//ListToMat;
 
 (*Fermion propagators*)
-	fermionPropT=Table[1/(t-i),{i,fermionMass}]//ListToMat;
-	fermionPropU=Table[1/(u-i),{i,fermionMass}]//ListToMat;
+	fermionPropT=Table[Prop[t,i],{i,fermionMass}]//ListToMat;
+	fermionPropU=Table[Prop[u,i],{i,fermionMass}]//ListToMat;
 	
 (*Group invariants from vector diagrams*)
-	CVS=Contract[gTensorS[[1,2]],vectorPropS . gTensorF[[3,4]],{{1,4}}]//OrderArray[#,1,2,3,4]&;
+	CSV=Contract[gTensorS[[1,2]], vectorPropS . gTensorF[[3,4]],{{1,4}}]//OrderArray[#,1,2,3,4]&;
+	CSVC=Contract[gTensorS[[1,2]], vectorPropS . gTensorF[[4,3]],{{1,4}}]//OrderArray[#,1,2,4,3]&;
+	
+(*Group invariants from Yukawa diagrams*)
+	CSy=Contract[\[Lambda]3Tensor[[1,2]], scalarPropS . YTensor[[3,4]],{{1,4}}]//OrderArray[#,1,2,3,4]&; 
+	
+	CTy=Contract[YTensor2[[1,3]] . fermionPropT , YTensor2[[2,4]],{{3,6}}]//OrderArray[#,1,3,2,4]&;
+	CUy=Contract[YTensor2[[1,4]] . fermionPropU , YTensor2[[2,3]],{{3,6}}]//OrderArray[#,1,3,4,2]&;
 	
-(*Group invariants from Yukawa diagrams*)	
-	CYS=Contract[\[Lambda]3Tensor[[1,2]],scalarPropS . YTensor[[3,4]],{{1,4}}]//OrderArray[#,1,2,3,4]&; 
-	CYT=Contract[YTensor2[[1,3]] . fermionPropT,YTensor2C[[2,4]],{{3,6}}]//OrderArray[#,1,3,2,4]&;
-	CYU=Contract[YTensor2[[1,4]] . fermionPropU,YTensor2C[[2,3]],{{3,6}}]//OrderArray[#,1,3,4,2]&;
+	CTyC=Contract[YTensor2[[1,3]] . fermionPropT , YTensor2C[[2,4]],{{3,6}}]//OrderArray[#,1,3,2,4]&;
+	CUyC=Contract[YTensor2[[1,4]] . fermionPropU , YTensor2C[[2,3]],{{3,6}}]//OrderArray[#,1,3,4,2]&;
+	
+	CTyCC=Contract[YTensor2C[[1,3]] . fermionPropT , YTensor2C[[2,4]],{{3,6}}]//OrderArray[#,1,3,2,4]&;
+	CUyCC=Contract[YTensor2C[[1,4]] . fermionPropU , YTensor2C[[2,3]],{{3,6}}]//OrderArray[#,1,3,4,2]&;
```

Comment:
You have introduce a bunch more contractions of tensors than the original code. Can you explain a bit why?